proto/src/error.rs

```diff
@@ -0,0 +1,31 @@
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum Error {
+    HexError(hex::FromHexError),
+    TooMuchData { expected: usize, got: usize },
+    InsufficientData { expected: usize, got: usize },
+}
```

Comment:
Should we make the name more specific here? Not sure if it is possible though.

Reply:
Ofcourse, what do you prefer? something like `DigestTooMuchData`?

Reply:
changed to `ParseError`, let me know if that is better.

Reply:
Yep - that works!